projects/libexif/exif_loader_fuzzer.cc

```diff
@@ -0,0 +1,31 @@
+#include <stdio.h>
+#include <stdint.h>
+#include <libexif/exif-loader.h>
+
+
+void content_func(ExifEntry *entry, void *user_data) {
+  char buf[10000];
+  exif_entry_get_value(entry, buf, sizeof(buf));
```

Comment:
Is there any way to get size of the entry first, then allocate a buffer of that size and only then get the value?

Reply:
Looks like `entry->size` might be pointing to that value, relying on that size when allocating a buffer seems to be a much more realistic API usage to me.

Reply:
The data in the entry is potentially expanded out when writing to the buffer because `exif_entry_get_value` is meant for human consumption. You can see an example of what it potentially writes here: https://github.com/libexif/libexif/blob/a881aa638711e87f340ea50783276b2ac62a837b/libexif/exif-entry.c#L961-L962

That means `entry->size` isn't a safe way to size the buffer. In fact, I can't find a way to know what the required size will be? Everything in the function uses `strncpy` or `snprintf` so it will just truncate the text to the buffer size though. Unless, of course, they have a bug 😄